neo4j-backup/templates/neo4j-cronjob.yaml

```diff
@@ -0,0 +1,98 @@
+{{- template "neo4j.backup.checkIfSecretExistsOrNot" . -}}
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: "{{ include "neo4j.fullname" . }}"
+  labels:
+    app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+    app.kubernetes.io/instance: {{ include "neo4j.fullname" . | quote }}
+    {{- include "neo4j.labels" $.Values.neo4j.labels | indent 4 }}
+spec:
+  schedule: {{ $.Values.neo4j.jobSchedule | default "* * * * *" | quote }}
+  concurrencyPolicy: Forbid
+  successfulJobsHistoryLimit: {{ $.Values.neo4j.successfulJobsHistoryLimit | default 3 }}
+  failedJobsHistoryLimit: {{ $.Values.neo4j.failedJobsHistoryLimit | default 1 }}
+  jobTemplate:
+    spec:
+      backoffLimit: {{ $.Values.neo4j.backoffLimit | default 6 }}
+      template:
+        metadata:
+          annotations:
+            {{- include "neo4j.annotations" $.Values.neo4j.podAnnotations | indent 12 }}
+          labels:
+            {{- include "neo4j.labels" $.Values.neo4j.podLabels | indent 12 }}
+        spec:
+          {{- if .Values.serviceAccountName }}
+          serviceAccountName: {{ .Values.serviceAccountName }}
+          {{- /* explicitly mount token because some service accounts disable automount-by-default and require explicit opt-in */}}
+          automountServiceAccountToken: true
+          {{- end }}
+          restartPolicy: Never
+          securityContext: {{ .Values.securityContext | toYaml  | nindent 12 }}
+          containers:
+            - name: graph-backup
+              image: {{ .Values.neo4j.image }}:{{ .Values.neo4j.imageTag }}
+              imagePullPolicy: Always
+              resources:
+                requests:
+                  ephemeral-storage: {{ .Values.resources.requests.ephemeralStorage | default "4Gi" }}
+                limits:
+                  ephemeral-storage: {{ .Values.resources.limits.ephemeralStorage | default "5Gi" }}
+              env:
+                - name: ADDRESS
+                  value: {{ required "You must specify a ADDRESS variable in values.yaml in the format <host>:<port> . Set it via --set backup.address" .Values.backup.address  | trim }}
```

Comment:
The IP / DNS of the DB could change over time. I think this should be something like `NEO4J_CLUSTER_NAME` and `NEO4J_NAMESPACE` then use k8s service discovery to locate the internal IP for backup.

Reply:
so the ADDRESS  is a combination of host:port which will be basically  ex: standalone-admin.default.svc.cluster.local:6362

IP will work too if the user wants to use that

Reply:
ok I still prefer something like;

```
backup:
     databaseAdminServiceName:
     databaseNamespace:
     databaseBackupPort: 6362
```

as this is much more self documenting

Reply:
IP should probably not be allowed as it frequently changes due to events such as auto scaling

Reply:
i dont think we should restrict much..there might be some scenarios where the dns is skewed ..the ip option should be their as it can act as a backup plan incase the dns is not resolving for some weird reason

Reply:
also we will need to ask for the domain (cluster.local) as this can change for different customers

Reply:
> so it will be somewhat like this
>
>backup:
>databaseAdminServiceName: standalone-admin
>databaseNamespace: default
>databaseBackupPort: 6362
>databaseClusterDomain: cluster.local

Yeah this sounds like a good compromise

Reply:
i would still be naming it as this to keep the IP support ...an extra additional "planb" for users just incase dns isn't working

backup:
databaseAdminServiceNameOrIP: standalone-admin
databaseNamespace: default
databaseBackupPort: 6362
databaseClusterDomain: cluster.local

when IP is provided namesapce and clsuter domain will be ignored or not needed.

Reply:
Yeah or alternatively 

```
backup:
databaseAdminServiceName: standalone-admin
databaseAdminServiceIP: x.x.x.x
databaseNamespace: default
databaseBackupPort: 6362
databaseClusterDomain: cluster.local
```
then raise an error if both are configured

Reply:
yeah this looks more clear...will go with this and update you back

Reply:
hey @ojhughes  this is implemented now...closing this conversation